### PR TITLE
PWX-33665: tests for ensuring hyperconvergence of KubeVirt VMs

### PR DESCRIPTION
**Pull request by pureneelesh**
**What type of PR is this?**
integration-test

**What this PR does / why we need it**:
Added 2 tests to verify stork and PX changes to return VMs to hyperconvergence after planned node maintenance done by OCP.

**Does this PR change a user-facing CRD or CLI?**:

no

**Is a release note needed?**:

no

**Does this change need to be cherry-picked to a release branch?**:

no

### CODE REVIEW

**Comment by pureneelesh** [comment]
Local run passed:
```
--- PASS: TestKubevirt (2167.46s)
    --- PASS: TestKubevirt/kubevirtDeployFedoraVMWithClonePVC (220.23s)
    --- PASS: TestKubevirt/kubeVirtHypercOneLiveMigration (301.05s)
    --- PASS: TestKubevirt/kubeVirtHypercTwoLiveMigrations (752.28s)
PASS

DONE 4 tests in 2196.153s
```

File: test/integration_test/kubevirt_hyperconv_test.go [hunk 1]
```diff
@@ -0,0 +1,499 @@
...
+	}
+	return mountTypeNFS, nil
+}
+
+func verifyVMStayedUp(t *testing.T, testState *kubevirtTestState) {
+	vmiUIDAfter, vmiPhaseAfter, transitionTimeAfter, err := getVMIDetails(testState.vmPod.Namespace, testState.vmiName)
+	require.NoError(t, err)
+	require.Equal(t, "Running", vmiPhaseAfter)
+	require.Equal(t, testState.vmiUID, vmiUIDAfter)
+	require.Equal(t, testState.vmiPhaseTransitionTime, transitionTimeAfter)
```
Comment (Rohit-PX) [inline test/integration_test/kubevirt_hyperconv_test.go:483]: Just curious whether this is expected to match every time. Is it not a function of the system load etc?

Comment (pureneelesh) [reply test/integration_test/kubevirt_hyperconv_test.go:483]: I verified that if a VM is stopped and started again, a new VMI object gets created with the same name (i.e. the UID will change). We are using that fact here to ensure that VM did not stop during our test. Let me know if that answers your question. I did not understand reference to system load.

File: test/integration_test/kubevirt_hyperconv_test.go [hunk 2]
```diff
@@ -0,0 +1,501 @@
...
+	// this test expects only one volume in the VM
+	require.Equal(t, len(vols), 1)
+	testState.volume = vols[0]
+
+	testState.apiVol, err = volumeDriver.InspectVolume(testState.volume.ID)
+	require.NoError(t, err, "Failed to inspect PV %s", testState.volume.ID)
+
+	testState.attachedNode, err = volumeDriver.GetNodeForVolume(testState.volume, cmdTimeout, cmdRetry)
+	require.NoError(t, err)
+	log.Infof("volume %s (%s) is attached to node %s",
```
Comment (pp511) [inline test/integration_test/kubevirt_hyperconv_test.go:170]: nit..Volume

File: test/integration_test/kubevirt_hyperconv_test.go [hunk 3]
```diff
@@ -0,0 +1,501 @@
...
+// Get mount type (nfs or bind) of the root disk volume. This function assumes that the volume name inside
+// the pod yaml is "rootdisk". e.g.
+//
+//	  volumes:
+//	  - name: rootdisk
+//	    persistentVolumeClaim:
+//		  claimName: fedora-communist-toucan
+func getVMRootDiskMountType(pod *corev1.Pod, apiVol *api.Volume) (string, error) {
+	podNamespacedName := pod.Namespace + "/" + pod.Name
+	logrus.Infof("checking the rootdisk mount type in pod %s", podNamespacedName)
```
Comment (pp511) [inline test/integration_test/kubevirt_hyperconv_test.go:430]: nit..Checking

File: test/integration_test/kubevirt_hyperconv_test.go [hunk 4]
```diff
@@ -0,0 +1,501 @@
...
+		return "", fmt.Errorf("failed to run command %v inside the pod %s", cmd, podNamespacedName)
+	}
+	var foundBindMount, foundNFSMount bool
+	for _, line := range strings.Split(output, "\n") {
+		if bindMountRE.MatchString(line) {
+			if foundBindMount || foundNFSMount {
+				return "", fmt.Errorf("multiple rootdisk mounts found: %s", output)
+			}
+			foundBindMount = true
+			logrus.Infof("found root disk bind mounted for VM pod %s: %s", podNamespacedName, line)
```
Comment (pp511) [inline test/integration_test/kubevirt_hyperconv_test.go:458]: nit..Found

File: test/integration_test/kubevirt_hyperconv_test.go [hunk 5]
```diff
@@ -0,0 +1,501 @@
...
+		}
+
+		if nfsMountRE.MatchString(line) {
+			if foundBindMount || foundNFSMount {
+				return "", fmt.Errorf("multiple rootdisk mounts found: %s", output)
+			}
+			foundNFSMount = true
+			logrus.Infof("found root disk nfs mounted for VM pod %s: %s", podNamespacedName, line)
```
Comment (pp511) [inline test/integration_test/kubevirt_hyperconv_test.go:466]: nit..Found

**Review by pp511 — APPROVED** [review]
This is really well written. 
Looks good.

**Pull request merged**